server/src/router/v1/index.ts

```diff
@@ -853,13 +854,218 @@ v1App.openapi(
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
+			[HttpStatusCodes.CONFLICT]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Conflict',
+			},
+		},
+	}),
+	async (c) => {
+		const { name } = c.req.valid('json');
+		const { major } = c.req.valid('param');
+
+		const newMajor = await db
+			.update(majors)
+			.set({ name })
+			.where(eq(majors.name, major))
+			.returning();
+		if (newMajor.length === 0) {
+			return c.json({ error: 'Major not updated' }, HttpStatusCodes.CONFLICT);
+		}
+		return c.text('', HttpStatusCodes.NO_CONTENT);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'delete',
+		path: '/majors/{major}',
+		tags: ['majors'],
+		summary: 'delete major',
+		middleware: [authMiddleWare('admin')],
+		request: {
+			params: z.object({
+				major: z.string(),
+			}),
+		},
+		responses: {
+			[HttpStatusCodes.NO_CONTENT]: {
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
+			[HttpStatusCodes.NOT_FOUND]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Conflict',
+			},
+		},
+	}),
+	async (c) => {
+		const { major } = c.req.valid('param');
+		
+		const newMajor = await db
+			.delete(majors)
+			.where(eq(majors.name, major))
+			.returning();
+
+		if (newMajor.length === 0) {
+			return c.json({ error: 'Major not deleted' }, HttpStatusCodes.NOT_FOUND);
+		}
+		return c.text('', HttpStatusCodes.NO_CONTENT);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'post',
+		path: '/majors',
+		tags: ['majors'],
+		summary: 'create major',
+		middleware: [authMiddleWare('admin')],
+		request: {
+			body: {
+				content: {
+					'application/json': {
+						schema: majorSchema,
+					},
+				},
+			},
+		},
+		responses: {
+			[HttpStatusCodes.CREATED]: {
+				content: {
+					'application/json': {
+						schema: majorSchema,
+					},
+				},
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
```

Comment:
I don't think this FORBIDDEN status will be triggered, should remove

Reply:
why won't forbidden be triggered. if non-admin tries to access the route manually or however, we should return 403

Reply:
i changed the one of the return codes to 403 in authMiddleware

Reply:
we didn't set that up, we can discuss further
for some projects, I only use 401 everything to avoid bruteforce and enumeration

Reply:
it's already setup. you already have the right condition in the middleware, we just have to return 403 instead of 401

Reply:
i don't know what you mean by avoiding bruteforce and enumeration. i think having the distinction is good. if someone gets 401 we can redirect to login, but if someone gets 403, we can redirect to their profile